django/contrib/contenttypes/views.py

```diff
@@ -64,7 +64,7 @@ def shortcut(request, content_type_id, object_id):
                 if field.remote_field and field.remote_field.model is Site:
                     try:
                         object_domain = getattr(obj, field.name).domain
-                    except Site.DoesNotExist:
+                    except (Site.DoesNotExist, AttributeError):
```

Comment:
I'd prefer something that makes it slightly more obvious where the exceptions are expected to happen:
```python
try:
    site = getattr(obj, field.name)
except Site.DoesNotExist:
   pass
if site is not None:
    object_domain = site.domain
```
This is a slightly different issue, but it's not clear if `Site.DoesNotExist` can actually happen (no tests fail if it's removed).